src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Policy/New-AzRecoveryServicesBackupPolicy.ps1

```diff
@@ -0,0 +1,186 @@
+    
+        [Parameter(DontShow)]
+        [ValidateNotNull()]
+        [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Runtime.SendAsyncStep[]]
+        # SendAsync Pipeline Steps to be prepended to the front of the pipeline
+        ${HttpPipelinePrepend},
+    
+        [Parameter(DontShow)]
+        [System.Uri]
+        # The URI for the proxy server to use
+        ${Proxy},
+    
+        [Parameter(DontShow)]
+        [ValidateNotNull()]
+        [System.Management.Automation.PSCredential]
+        # Credentials for a proxy server to use for the remote call
+        ${ProxyCredential},
+    
+        [Parameter(DontShow)]
+        [System.Management.Automation.SwitchParameter]
+        # Use the default credentials for the proxy
+        ${ProxyUseDefaultCredentials},
+            
+        [Parameter(Mandatory=$false)]
+        [Nullable[System.Boolean]]
+        ${MoveToArchiveTier},
+        
+        [Parameter(Mandatory=$false)]
+        [ValidateSet('TierRecommended', 'TierAfter')]
+        [string]
+        ${TieringMode},
+        
+        [Parameter(Mandatory=$false)]
+        [Nullable[int]]
+        ${TierAfterDuration},
+
+        [Parameter(Mandatory=$false)]
+        [ValidateScript({$_ -ge 1 -and $_ -le 5}, ErrorMessage = "The value of SnapshotRetentionDurationInDays must be between 1 and 5.")]
+        [Nullable[int]]
+        ${SnapshotRetentionDurationInDays}
+        
+    )
+
+    process
+    {   
+        #get datasource type
+        $BackupManagementType = $policy.BackupManagementType
+        $WorkloadType = $policy.WorkLoadType
+        $DataSourceType = Get-DataSourceType -BackupManagementType $BackupManagementType -WorkloadType $WorkloadType
+        $manifest = LoadManifest -DatasourceType $DatasourceType.ToString()
+
+        if($SnapshotRetentionDurationInDays -ne $null)
+        {
+            $policy.instantRpRetentionRangeInDay=$SnapshotRetentionDurationInDays
+            if(($policy.SchedulePolicy.ScheduleRunFrequency -eq "Weekly") -and ($policy.instantRpRetentionRangeInDay -ne 5))
+            {
+                $errormsg="SnapshotRetentionDuration value must be 5 for backup policy with frequency of schedule as Weekly"
+                throw $errormsg
+            }
+        }
+
+        if($MoveToArchiveTier -ne $null)
+        {
+            if($policy.BackupManagementType -eq "AzureIaasVM")
+            {
+                 $tieringdetails = $policy.TieringPolicy.AdditionalProperties.ArchivedRP
+            }
+            elseif($policy.BackupManagementType -eq "AzureWorkload")
+            {
+                $FullBackupPolicy =  $policy.SubProtectionPolicy | where { $_.PolicyType -match "Full" }
+                $Index = $policy.SubProtectionPolicy.IndexOf($FullBackupPolicy)
+                $tieringdetails =$policy.SubProtectionPolicy[$Index].TieringPolicy.AdditionalProperties.ArchivedRP
+            }
+            
+            if($manifest.IsTieringSupported -ne "True")
```

Comment:
will this be a string comparison or boolean $true

Reply:
I have used it as string

Reply:
yes I think it shoud be boolean if the values are $true/$false